docs/src/main/asciidoc/_admin-guide/subsystem-configuration/MicroProfile_Health.adoc

```diff
@@ -79,28 +107,44 @@ created by the `add-user` script. For example:
 $ curl -v --digest -u myadminuser:myadminpassword http://localhost:9990/health
 < HTTP/1.1 200 OK
 ...
-{"outcome":"UP","checks":[]}
+{"status":"UP","checks":[{"name":"empty-liveness-checks","status":"UP"},{"name":"server-state","status":"UP","data":{"value":"running"}},{"name":"boot-errors","status":"UP"},{"name":"deployments-status","status":"UP"},{"name":"empty-readiness-checks","status":"UP"}]}
 ----
 
 If the authentication fails, the  server will reply with a `401 NOT AUTHORIZED` response.
 
-=== Global Status when probes are not defined
+=== Default Server Procedures
+
+WildFly provides some readiness procedures that are checked to determine if the application server is ready to serve requests:
+
+* `boot-errors` checks that there were no errors during the server boot sequence
+* `deployments-status` checks that all deployments were deployed without errors
+* `server-state` checks that the server state is `running`
+* `empty-readiness-checks` determines the status when there are no readiness check procedures deployed to the server. The outcome of this procedure is determined by the `empty-readiness-checks-status` attribute. If the attribute is
+   `UP` (by default), the server can be ready when there are no readiness checks in the deployments. Setting the `empty-readiness-checks-status` attribute to `DOWN` will make this procedure fail when there are no readiness checks in the deployments.
+
+If a deployment does not provide any readiness checks, WildFly will automatically add one for each deployment (named `ready-<deployment name>`) which always returns `UP`.
```

Comment:
> > @jmesnil Is `ready-<deployment name>` subset of `deployments-status` ? Can we get rid of that? `ready-<deployment name> always returns UP` means it does not reflect deployment errors?
> 
> They serve different purpose.
> `ready-<deployment name>` is added by WildFly for _each_ deployment if the deployment does not contain any readiness check. 
And my point is if we can't use deployments-status for this purpose?
 
> It only returns `UP` for this deployment. However if the deployment provides a readiness check, this one will be used to determine if the deployment is ready or not.
> `deployments-status` checks that _all deployments_ have been deployed without error.

few notes:
MP should serve one deployment.
I do have detailed info in `deployments-status` about deployments so I am not loosing this information per deployment anyway.
```
            {
                "name" => "deployments-status",
                "status" => "UP",
                "data" => {"microprofile-config.war" => "OK"}
            },
```
 
Can it happen  `ready-<deployment name>` is UP but `deployments-status` is DOWN (with one deployment)? Can it happen  `ready-<deployment name>` is DOWN but `deployments-status` is UP (with one deployment)? In other words do they have still same result?

> 
> > Is it possible `empty-readiness-checks-status` make work with `deployments-status` ?
> > This is somehow backward incompatible change, but would IMO simplify whole logic of health check.
> 
> I'm not sure it's a good idea. The goal of this RFE is to provides the _same_ checks that the EAP probes provide using the health subsystem instead of external scripts.
> Any change of behaviour would have to be in a way that the semantics of the "healthiness" of the app server are preserved regardless of the way they are checked (external scripts and health subsystem).


I am not sure if this should be added under this RFE (mainly because of backward compatibility questions), but thinking about it with fresh context. Just it was hard to me to understand all combinations. And I am thinking how to simplify this unnecessary complexity. 

Also about `empty-readiness-checks-status`. Cant problem which is `empty-readiness-checks-status` fixing happen to `deployments-status` ,`boot-errors`, `server-state`? I mean that :check is reporting at beginning UP, but after some of this check finished it becomes DOWN? Don't we need to force that somehow globally? To report DOWN, until all checks finished? (I don't know if that is feasible)

Reply:
WildFly supports multiple deployments. We should strive to take that into account when providing MP features if that does not conflict with the spec.

* If the deployment has no readiness checks*, `ready-<deployment name>` and `deployments-status` (assuming there is a single deployment) will have the same outcome
* If the deployment has readiness checks, the outcome will be determined by them.

> Cant problem which is empty-readiness-checks-status fixing happen to deployments-status ,boot-errors, server-state?

There is a short window where it is true: when `server-state` is `UP` (the server is running) and the deployment are not finished (deployments-status will not report any errors as the deployments are processed.
This windows can be alleviated by setting `empty-readiness-checks-status` to `DOWN` so that the whole server readiness will be `DONE` until deployments have been successfully deployed.